# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`, `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c = 111
d = 402
print(((a + b)*c)/d)

85.8731343283582


Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [4]:
print(11283 % 89)

69


3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [3]:
print(11283 // 89)

126


4. Запишіть у змінну `name` ваше імʼя.

In [7]:
name = "Sofiia"

5. Запишіть у змінну `surname` ваше прізвище.

In [8]:
surname = "Nizovtsova"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [9]:
name_surname = name + " " + surname
print(name_surname)

Sofiia Nizovtsova


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [10]:
print(name_surname.upper())

SOFIIA NIZOVTSOVA


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [11]:
print(name_surname[2])

f


Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [22]:
print(surname[0]) # first letter, if we know the WHOLE surname

len(surname)


N


10

In [25]:
print(surname[9]) # last letter, if we know the WHOLE surname

a


In [27]:
print(surname.find("N")) #first letter and its index number
print(surname.find("a")) #last letter and its index number, if we dont know the length of the surname

0
9


In [31]:
print(surname[0], surname[9])

N a


Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [32]:
len(surname)

10

Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [91]:
capital_Mad = "Antananarivo"
upper_a_capital_Mad = capital_Mad.count('A') # with 'find' it doesnt work, only works with count
lower_a_capital_Mad = capital_Mad.find('a', 0, 11)
print (upper_a_capital_Mad + lower_a_capital_Mad)


4


In [78]:
#OR2
capital_Mad2 = "Antananarivo"
lower_a_capital_Mad2 = capital_Mad2.lower().count('a', 0, 11)
print(lower_a_capital_Mad2)


4


In [83]:
#OR3
num_a_capital_Mad3 = capital_Mad[0:11].count('A') + capital_Mad[0:11].count('a')
print(num_a_capital_Mad3)

4


Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [87]:
my_string = 'The quick brown fox jumps over the lazy dog!!!'

print(my_string.startswith('The'))
print(my_string.endswith('dog!'))


True
False


In [96]:
#OR
if my_string.startswith('The') == True:
    print('Ей! Цей рядок починається зі слова "The"')
else:
    print('На жаль, цей рядок не починається зі слова "The"')
    
if my_string.endswith('dog!') == True:
    print('Цій рядок закінчується словом "dog!"')
else:
    print('На жаль, цей рядок не закінчується словом "dog!"')

Ей! Цей рядок починається зі слова "The"
На жаль, цей рядок не закінчується словом "dog!"


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [129]:
separated_str = (my_string.split())
print(separated_str)

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']


In [127]:
print('_'.join(separated_str))

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`